phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -488,7 +490,55 @@ public void rollback(PhoenixConnection connection) throws SQLException {
         numRows = 0;
     }
     
+    private Set<Integer> getOrderOfUncommittedStatements() {
+    	Set<Integer> result = newHashSet();
+    	for (Map<ImmutableBytesPtr, RowMutationState> rowMutations : mutations.values()) {
+    		for (RowMutationState rowMutationState : rowMutations.values()) {
+    			result.addAll(rowMutationState.getOrderOfStatementsInConnection());
+    		}
+    	}
+    	return result;
+    }
+    
     @Override
     public void close() throws SQLException {
     }
+    
+    public static class RowMutationState {
+    	private Map<PColumn,byte[]> columnValues;
+    	private Set<Integer> orderOfStatementsInConnection;
```

Comment:
Hmm. I guess I'm not too concerned about memory here, since the size of this set is going to be proportional to the number of executed statements on a connection, which is going to be thousands at the high end for realistic use cases IMHO.